File: Go/01.Go-Basic-February-2022/09.advancedLoops-Lab/07.minNumber.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func main() {
	var min int = math.MaxInt32

	var input string
	fmt.Scanln(&input)

	for input != "Stop" {

		number, _ := strconv.Atoi(input)

		if number < min {
			min = number
		}

		fmt.Scanln(&input)
	}

	fmt.Println(min)
}

File: JavaScript/01.JavaScript-Basic-February-2021/09.whileLoop-Lab/07.minNumber.js
function minNumber(input) {
    let index = 0;
    let current = input[index];
    let min = Number.MAX_SAFE_INTEGER;

    while (current != "Stop") {
        let currentNumber = Number(current);
        if (min > currentNumber) {
            min = currentNumber;
        }
        index++;
        current = input[index];
    }
    console.log(min);
}

minNumber(["100", "99", "80", "70", "Stop"]);
minNumber(["-10", "20", "-30", "Stop"]);
minNumber(["45", "-20", "7", "99", "Stop"]);
minNumber(["999", "Stop"]);
minNumber(["-1", "-2", "Stop"]);